tests/base.py

```diff
@@ -0,0 +1,108 @@
+import os
+import pathlib
+import random
+import subprocess
+import tempfile
+import time
+import unittest
+
+import rpyc
+
+from .utils import debug_target
+
+COVERAGE_DIR = os.getenv("COVERAGE_DIR", "")
+GEF_PATH = pathlib.Path(os.getenv("GEF_PATH", "gef.py")).absolute()
+RPYC_GEF_PATH = GEF_PATH.parent / "scripts/remote_debug.py"
+RPYC_HOST = "localhost"
+RPYC_PORT = 18812
+RPYC_SPAWN_TIME = 1.0
+RPYC_MAX_REMOTE_CONNECTION_ATTEMPTS = 5
+
+
+class RemoteGefUnitTestGeneric(unittest.TestCase):
+    """
+    The base class for GEF test cases. This will create the `rpyc` environment to programmatically interact with
+    GDB and GEF in the test.
+    """
+
+    def setUp(self) -> None:
+
+        attempt = RPYC_MAX_REMOTE_CONNECTION_ATTEMPTS
+        while True:
+            try:
+                #
+                # Port collisions can happen, allow a few retries
+                #
+                self._coverage_file = None
+                self.__setup()
+                break
+            except ConnectionRefusedError:
+                attempt -= 1
+                if attempt == 0:
+                    raise
+                time.sleep(0.2)
+                continue
+
+        self._gdb = self._conn.root.gdb
```

Comment:
So root is the scope that the gef.py file loads? I can't think of a better name, but maybe you can?

Reply:
`root` is the namespace of the `RemoteGefUnitTestGeneric` 
Also the name comes from rpyc, not me